script.js

```diff
@@ -29,4 +49,126 @@ function operate(operator, a, b) {
         case "/":
             return divide(a,b);
     }
-}
\ No newline at end of file
+}
+
+function calculateFromReversePolishNotation(rpn) {
+    workingRPN = [...rpn]
+    while (workingRPN.length > 1) {
+        [num1, num2, mathSymbol] = workingRPN.splice(0,3)
+        runningResult = operate(mathSymbol, num1, num2)
+        if (runningResult == DIVIDE_BY_ZERO_ERROR) {
+            return DIVIDE_BY_ZERO_MESSAGE
+            
+        }
+        workingRPN.unshift(runningResult)
+    }
+    return runningResult;
+}
+
+function updateInputDisplay(list) {
+    const displayScreen = document.querySelector(`.input`);
+    displayScreen.textContent = list.join("");
+}
+
+function addToInputDisplay(character, list) {
+    list.push(character);
+    updateInputDisplay(list);
+}
+
+function updateResultDisplay(value) {
+    const displayScreen = document.querySelector(`.results`);
+    displayScreen.textContent = value;
+}
+
+function clearAll() {
+    buffer = [];
+    currentlyHeldOperator = null;
+    inputsForDisplay = [];
+    reversePolishNotation = [];
+    runningResult = 0;
+    updateInputDisplay([])
+    updateResultDisplay(INITIAL_RESULT_DISPLAY_VALUE)
+}
+
+function doesArrayContainOperator(array) {
+    const [possibleOperator] = array;
+    return OPERATORS.includes(possibleOperator);
+}
+
+// After calculation is done, clicking on another button will clear the previous information and also the display
+const buttons = document.querySelectorAll("button");
+buttons.forEach( (button) => {
+    button.addEventListener("click", (e) => {
+        if (currentlyHeldOperator === FINISHED_CALCULATION) {
+            clearAll()
+        }
+    })
+})
+
+const clearButton = document.querySelector("#clear");
+clearButton.addEventListener("click", (e) => {
+    clearAll();
+})
+
+const digitButtons = document.querySelectorAll(".digit");
+digitButtons.forEach( (digitButton) => {
+    digitButton.addEventListener("click", (e) => {
+        const digit = e.target.getAttribute("data-value")
+        if (doesArrayContainOperator(buffer)) {
+            buffer = [digit]
+            addToInputDisplay(digit, inputsForDisplay);
+        } else if (buffer.length == 0 || NUMBERS.includes(buffer[buffer.length-1])) {
```

Comment:
I think, since the first "if" statement checks if there's an operator in the buffer, that this being an "else" might work. 
Since the 3 possibilities for the buffer are:
1, Length = 0 (so buffer.push)
2, Last character is a number (so buffer.push)
3, Buffer is an operator (so buffer = digit)

I prefer to have more specific if statements to constrain behavior (in case code changes and the last character maybe becomes a decimal point for example).

Reply:
Updated logic to check if the buffer contains number using "doesArrayContainNumber" function.